frontend/connection/src/WebsocketConnection.tsx

```diff
@@ -104,6 +104,15 @@ export interface Args {
    * scenario).
    */
   onHostConfigResp: (resp: IHostConfigResponse) => void
+
+  /**
+   * Enables host to bypass waiting for health/host-config endpoint responses
+   * in establishing the initial websocket connection. When true, the connection
+   * state machine will connect to the websocket immediately without waiting for
+   * the initial ping cycle to complete. Health and host-config pings continue to run
+   * asynchronously in the background for error handling and configuration.
```

Comment:
One potential issue with supporting host config configuration after the websocket has been established: If the host config gets loaded after the app has already started rendering, it could cause potential unexpected behaviour in this first run. Most of the host-config configuration somewhat relies on the aspect that it's configured correctly before the first element gets rendered. However, it is likely that this doesn't cause too much trouble in practice. Does SiS need to configure Streamlit through the window variable and the endpoint return?

Reply:
True, this is a concern - particularly with fields like `resourceCrossOriginMode` - and I think the intention is to eventually migrate away from having the `host-config` endpoint.
For now, we will recommend passing all fields through window var (particularly now that it has been secured and these take precedence).